### PR TITLE
Fix AMI percent input bug #172688620

### PR DESCRIPTION
**Pull request by james-wills-sf**
[pivotal ticket #172688620](https://www.pivotaltracker.com/story/show/172688620)

[review app link](https://dahlia-lap-full-pr-352.herokuapp.com)

## Summary
This change makes the supplemental application form submit properly when
the AMI percent input text has a percent sign in it. Additionally, this
change makes the behavior of the percent input closer to that of the
currency input, where a percent sign is automatically added after a
user defocuses the view, and the input only allows numbers (no decimal
points) so a failure case ends up being pretty much impossible.

## Screenshot (before): page error when submitting with percent sign in input
- 2020-05-06 at 5 14 42 PM

## Screencast (after): AMI percentages input
Note that the user is only allowed to type digits [0-9], if you type anything else it won't type anything. This is the same behavior the currency inputs have
AMIpercentage

## Screencast (after): Currency input (just to show that this didn't break)
currencyform

### CODE REVIEW

File: app/javascript/utils/form/final_form/Field.js
Comment (james-wills-sf) [inline app/javascript/utils/form/final_form/Field.js:106]: This is basically just copied and pasted from `CurrencyField`, except for the parse and format props in `Field`.

File: app/javascript/utils/formUtils.js
Comment (james-wills-sf) [inline app/javascript/utils/formUtils.js:24]: fix issue where `0` didn't trigger formatting in currency. Now it does

**Review by akegan — COMMENTED** [review]
One quick note from QA: looks like you can't type in "." so you can't do fractional percents. I'm not sure if they round or not. will ask in slack.

**Comment by akegan** [comment]
PR otherwise looks great! should be good to go once we resolve the int vs float question

**Comment by james-wills-sf** [comment]
@akegan I added a commit that updates the form to accept float values if you want to take another look! 

AMIPercentFloat

**Review by akegan — APPROVED** [review]
Looks good! Once you merge you can pull lastest master into the release branch. Master also contains https://github.com/SFDigitalServices/sf-dahlia-lap/pull/349 which I don't think is on the release branch but is related so I think we can pull both into the release branch

**Pull request merged**
